### PR TITLE
GIX-1219: Add api function to get proposal

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

User sees the details of a proposal for an SNS project.

This PR: Add api function the get the proposal by id

# Changes

* New api function `queryProposal`.

# Tests

* Test new api function.

### CODE REVIEW

**Comment by lmuntaner** [comment]
@dskloetd please review

File: frontend/src/lib/api/sns-governance.api.ts [hunk 1]
```diff
@@ -549,3 +549,28 @@ export const queryProposals = async ({
...
+    identity,
+    rootCanisterId: rootCanisterId.toText(),
+    certified,
+  });
+
+  const proposal = await getProposal({ proposalId });
+
+  logWithTimestamp(`Getting proposal ${proposalId.id} call complete.`);
+  return proposal;
+};
```
Comment (dskloetd) [inline frontend/src/lib/api/sns-governance.api.ts:576]: I see a lot of duplication, would it make sense to have a helper that takes the name of the api function and:
1. Logs the start
2. Gets the function
3. Calls the function
4. Logs the end
5. Returns the result

**Review by dskloetd — APPROVED** [review]

File: frontend/src/lib/api/sns-governance.api.ts [hunk 2]
```diff
@@ -549,3 +549,28 @@ export const queryProposals = async ({
...
+
+  const { getProposal } = await wrapper({
+    identity,
+    rootCanisterId: rootCanisterId.toText(),
+    certified,
+  });
+
+  const proposal = await getProposal({ proposalId });
+
+  logWithTimestamp(`Getting proposal ${proposalId.id} call complete.`);
```
[the remark this answers is not in the document]
Comment (lmuntaner) [reply frontend/src/lib/api/sns-governance.api.ts:574]: Why? we don't catch errors at this level.

File: frontend/src/lib/api/sns-governance.api.ts (hunk 1 above)
Comment (lmuntaner) [reply frontend/src/lib/api/sns-governance.api.ts:576]: I think that this is easy to read and to maintain. A little bit of code duplication is not always bad.

File: frontend/src/lib/api/sns-governance.api.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (dskloetd) [reply frontend/src/lib/api/sns-governance.api.ts:574]: finally is orthogonal to catching.

try {
  somethingThatMightThrow();
} finally {
  cleanup();
}

[the remark this answers is not in the document]
Comment (lmuntaner) [reply frontend/src/lib/api/sns-governance.api.ts]: done

**Pull request merged**